scripts/benchmark/checks/deploy_erc20.py

```diff
@@ -0,0 +1,167 @@
+import logging
+from typing import (
+    Tuple
+)
+
+from web3 import (
+    Web3
+)
+from web3.utils.transactions import (
+    TRANSACTION_DEFAULTS
+)
+
+from eth_utils import (
+    encode_hex,
+    decode_hex,
+    to_int,
+)
+
+import solc
+
+from evm.constants import (
+    CREATE_CONTRACT_ADDRESS
+)
+from evm.chains.base import (
+    Chain,
+)
+from evm.rlp.blocks import (
+    BaseBlock,
+)
+
+from .base_benchmark import (
+    BaseBenchmark,
+)
+from utils.chain_plumbing import (
+    FUNDED_ADDRESS,
+    FUNDED_ADDRESS_PRIVATE_KEY,
+    get_all_chains,
+)
+from utils.reporting import (
+    DefaultStat,
+)
+from utils.tx import (
+    new_transaction,
+)
+
+
+EXPECTED_TOTAL_SUPPLY = 10000000000000000000000
+FIRST_TX_GAS_LIMIT = 1400000
+SECOND_TX_GAS_LIMIT = 22000
+
+SOLIDITY_SRC_FILE = './scripts/benchmark/contract_data/erc20.sol'
+
+
+class DeployErc20(BaseBenchmark):
+
+    def __init__(self, num_blocks: int = 100, num_tx: int = 2) -> None:
+        super().__init__()
+
+        self.num_blocks = num_blocks
+        self.num_tx = num_tx
+
+        compiled_sol = solc.compile_files([SOLIDITY_SRC_FILE])
+        self.contract_interface = compiled_sol['{}:SimpleToken'.format(SOLIDITY_SRC_FILE)]
+
+        # Stop Web3 from trying to call out to external APIs
+        TRANSACTION_DEFAULTS['gas'] = lambda web3, tx: 0
+        TRANSACTION_DEFAULTS['gasPrice'] = lambda web3, tx: 0
```

Comment:
Global monkeypatches creep me out. Also, we really shouldn't have to reach into `utils` from another package.

If we pass in `{'gas': 0, 'gasPrice': 0}` to the `buildTransaction()` calls, that should prevent web3 from trying to generate defaults. Does anywhere else fail when you remove these lines?

Reply:
Ah, will try that!

Reply:
That worked. I wasn't happy with monkeypatching either. Happy to see there's a better way to use the API to avoid that :+1: